src/Lachain.Core/RPC/HTTP/Web3/BlockchainServiceWeb3.cs

```diff
@@ -439,42 +456,30 @@ private JArray GetEventsByTransactionHash(string txHash)
         }
 
         [JsonRpcMethod("eth_getLogs")]
-        private JArray GetLogs(JObject opts)
+        public JArray GetLogs(JObject opts)
         {
             var fromBlock = opts["fromBlock"];
             var toBlock = opts["toBlock"];
             var address = opts["address"];
             var topicsJson = opts["topics"];
             var blockhash = opts["blockHash"];
 
-            if (!(fromBlock is null) && !(toBlock is null) && !(blockhash is null))
+            if ((!(fromBlock is null) || !(toBlock is null)) && !(blockhash is null))
                 throw new Exception(
                     "If blockHash is present in in the filter criteria, then neither fromBlock nor toBlock are allowed.");
 
-            var start = (ulong)0;
+            ulong ?start;
+            start = _blockManager.GetHeight();
             if (!(fromBlock is null))
             {
-                if (((string)fromBlock!).StartsWith("0x"))
```

Comment:
If we use this functionality,  we should check if we can parse hex strings starting with "0x" and strings that are not started with special prefix. Add this logic to the GetBlockNumberByTag

Reply:
It is already implemented in Lachain.Utility.Utils.HexUtils.HexToUlong(this string buffer)